### PR TITLE
fix(web-app): override carbon overlay z-index

### LINKED ISSUE #249: Z index issue with the side nav overlay at medium breakpoint
### Browser

_No response_

### Description

We need to fix our `z-index`s to not conflict with the Carbon side nav overlay. Either reduce our z-index values and document those, or increase the z-index for Carbon's side nav overlay.

image

### Steps to reproduce

Medium breakpoint, catalog page, open side nav.

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #249

### CODE REVIEW

File: services/web-app/layouts/layout/layout.module.scss [hunk 1]
```diff
@@ -14,4 +14,7 @@
   // position for PageBreadcrumb
   position: relative;
   margin: $spacing-09 0 0;
+  :global(.cds--side-nav__overlay.cds--side-nav__overlay-active) {
+    z-index: 10;
```
Comment (mattrosno) [inline services/web-app/layouts/layout/layout.module.scss:18]: How'd you land on 10? Maybe just a personal preference, but I like to add a comment when I have to use a z-index besides -1, 0, or 1.

**Review by mattrosno — DISMISSED** [review]

File: services/web-app/layouts/layout/layout.module.scss (hunk 1 above)
Comment (francinelucca) [reply services/web-app/layouts/layout/layout.module.scss:18]: we're currently using 0,1 & 2 in our custom components in the web-app, I just needed a number that was higher than that, figured I'd add some wiggle room.
Happy to change it to lower if you think necessary!
image

Comment (mattrosno) [reply services/web-app/layouts/layout/layout.module.scss:19]: @francinelucca I'd either go 3 and add a comment why, or maybe use a Carbon z-index. I don't know how/where these are used. https://github.com/carbon-design-system/carbon/blob/48deedf2fb7a343594ed844d6c852c9108a3eb53/packages/components/src/globals/scss/_layout.scss#L14

Comment (mattrosno) [reply services/web-app/layouts/layout/layout.module.scss:19]: Oh, that's a private mixin. I don't know if Carbon has z-index's as part of the public API. I'd go 3.

Comment (mattrosno) [reply services/web-app/layouts/layout/layout.module.scss:19]: Oh again, but the function `z` is public. https://github.com/carbon-design-system/carbon/blob/48deedf2fb7a343594ed844d6c852c9108a3eb53/packages/components/src/globals/scss/_layout.scss#L35

**Review by mattrosno — APPROVED** [review]

**Pull request merged**